src/utils/mtev_confstr.c

```diff
@@ -169,6 +169,12 @@ mtev_confstr_parse_duration(const char *input, uint64_t *output,
     if(unit_str == input)
       return MTEV_CONFSTR_PARSE_ERR_FORMAT;
 
+    /* remove space(s) between number and unit string */
+    while(*unit_str && !isalpha(*unit_str)){
+      if(isdigit(*unit_str)) return MTEV_CONFSTR_PARSE_ERR_FORMAT;
```

Comment:
You don't need to check for a digit here if you just check for spaces in the initial loop. I'd recommend removing this line and handling all the error parsing after all the spaces have been skipped.

Reply:
```    /* remove space(s) between number and unit string */
    while(*unit_str && isspace(*unit_str))
      unit_str++;```